integration/singularity/store.go

```diff
@@ -282,11 +291,10 @@ func (s *Store) Start(ctx context.Context) error {
 		}
 	}
 
-	s.closed.Add(1)
-	go s.runPreparationJobs()
+	s.cleanupScheduler.start(ctx)
 
 	s.closed.Add(1)
```

Comment:
Instead of using a WaitGroup here, it is simpler to wait on a `chan struct{}` to close.

Reply:
hmmmm, I somewhat disagree, I default to wait group because if any other goroutines are added to a struct in the future that need to be closed, chan struct would no longer work. the difference in amount of lines of code is 1, and there is practically no performance difference for this use case. but if you still think it's better to use chan struct here, give me the word and i will be happy to change it

Reply:
Fine wither way.
If there may be others to wait for, then leave as WaitGroup.